uint8 sid
uint16 target_position
